tests/testthat/test-type-date-time.R

```diff
@@ -84,10 +92,16 @@ test_that("dates become midnight in date-time tzone", {
   date2 <- vec_cast(date1, datetime)
   expect_equal(attr(date2, "tzone"), "Pacific/Auckland")
   expect_equal(format(date2, "%H:%M"), "00:00")
+
+  datetime <- as.POSIXlt(datetime)
+  date2 <- vec_cast(date1, datetime)
+  expect_equal(attr(date2, "tzone")[1], "Pacific/Auckland")
```

Comment:
Can you add a comment here reminding us that the tz attribute for POSIXlt can have three components (time zone name, abbreviated name, abbreviated DST name).

I'd also recommend using `[[` instead of `[` to emphasis that we want a single value back